src/node/blockstorage.cpp

```diff
@@ -885,6 +886,7 @@ void ThreadImport(ChainstateManager& chainman, std::vector<fs::path> vImportFile
                 StartShutdown();
                 return;
             }
+            fReindexChainState = false;
```

Comment:
In commit "node: add fReindexChainState flag to node" (8898261ac03945e559c98a0abcb9455942ab4775)

I understand what this are doing with the new `fReindexChainState` global is similar to what happens with the existing `fReindex` global, but I think what happens with the `fReindex` global is not ideal. I think it's unnecessarily fragile how the value starts off false, then switches to true, then switches back to false again, and think a one way latch would be better. I think it's bad the the variable doesn't have any straightforward meaning but is some combination "was reindexing requested?" and "is reindexing in progress?". Also in this commit, `fReindexChainState` is different local variables subtly replaced by a global so you can't tell just looking at a diff whether all the existing `fReindexChainState` references were updated correctly.

I'd suggest leaving `fReindexChainState` alone here, and adding a simpler `atomic_bool g_indexes_ready_to_sync = false` global variable, and setting it to true here, and in the normal code path. I think this would make the diff and the overall initialization sequence simpler and easier to understand.

Reply:
That makes sense. I dropped the commit that made `fReindexChainState` a global and added `g_indexes_ready_to_sync` as suggested.